src/expr/src/scalar/func.rs

```diff
@@ -5752,10 +5751,13 @@ impl VariadicFunc {
             }
             ArrayToString { .. } => ScalarType::String.nullable(true),
             ListCreate { elem_type } => {
-                soft_assert!(
-                    input_types.iter().all(|t| t.scalar_type.base_eq(elem_type)),
-                    "Args to ListCreate should have types that are compatible with the elem_type"
-                );
+                // commented out to work around
```

Comment:
Can you please explain this more -- why is this assert firing? Why is it okay to ignore it?

#9008 doesn't seem to have any description.

Reply:
I will have to defer to @asenac as he is working on it actively.

Reply:
It doesn't have a description because it wasn't meant to be shared yet. The description of the problem is in https://github.com/MaterializeInc/materialize/issues/8963.